Partindo do ponto em que os dados já estão limpos, chegamos à etapa de "exploração dos dados". Primeiramente, pretendo entender quais são as colunas, que dados elas trazem e suas necessidades para a análise como um todo. Para isso podemos usar alguns comandos:

In [1]:
import numpy as np
import pandas as pd

df = pd.read_csv("../../data/processed/olist_super_dataset.csv")

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 109921 entries, 0 to 109920
Data columns (total 27 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   order_id                       109921 non-null  object 
 1   customer_id                    109921 non-null  object 
 2   order_status                   109921 non-null  object 
 3   order_purchase_timestamp       109921 non-null  object 
 4   order_approved_at              109906 non-null  object 
 5   order_delivered_carrier_date   109920 non-null  object 
 6   order_delivered_customer_date  109921 non-null  object 
 7   order_estimated_delivery_date  109921 non-null  object 
 8   tempo_entrega_dias             109921 non-null  float64
 9   atraso_entrega                 109921 non-null  bool   
 10  customer_unique_id             109921 non-null  object 
 11  customer_zip_code_prefix       109921 non-null  int64  
 12  customer_city                 

Como podemos observar, temos muitas ocorrências/registros para analisar. Isso traz os dois lados da moeda, tanto a abrangência dos dados em larga escala quanto o poder computacional necessário para trabalhar com todos esses dados...

Outra coisa à se observar é o fato de termos algumas linhas nulas, conforme vemos em:

In [ ]:
tot_lines = len(df.index)
null_lines = df.isnull().any(axis=1).sum()

print(f"""Quantidade de linhas nulas: {null_lines}
Quantidade de células nulas: {df.isnull().sum().sum()}
Porcentagem aem relação ao total: {(null_lines*100)/tot_lines:.2f}%""")

Quantidade de linhas nulas: 2360
Quantidade de células nulas: 5437
Porcentagem representada em relação ao total: 2.15%


Com essa informação, sabemos que algumas linhas possuem mais de uma célula nula. Agora temos de decidir o que fazer com essas linhas... Eu irei optar por retirá-las - uma vez que elas representam apenas 2.15% do total. Para retirá-las podemos usar a abordagem:

In [9]:
df.dropna(axis=0, how='any', inplace=True)

tot_lines = len(df.index)
null_lines = df.isnull().any(axis=1).sum()

print(f"""Quantidade de linhas nulas: {null_lines}
Quantidade de células nulas: {df.isnull().sum().sum()}
Porcentagem aem relação ao total: {(null_lines*100)/tot_lines:.2f}%""")

Quantidade de linhas nulas: 0
Quantidade de células nulas: 0
Porcentagem aem relação ao total: 0.00%


In [4]:
df.describe()

,tempo_entrega_dias,customer_zip_code_prefix,order_item_id,price,freight_value,product_description_lenght,product_photos_qty,payment_value,review_score,receita_liquida
count,109921.000000,109921.000000,109921.000000,109921.000000,109921.000000,108389.000000,108389.000000,109918.000000,109099.000000,109921.000000
mean,11.921607,35136.419401,1.198042,116.697762,19.748378,786.293249,2.210270,174.886274,4.083045,136.446140
std,8.808581,29905.228972,0.705973,156.221345,14.343424,650.232743,1.721581,224.799408,1.344884,162.511046
min,1.000000,1003.000000,1.000000,0.850000,0.000000,4.000000,1.000000,9.590000,1.000000,6.080000
25%,6.000000,11250.000000,1.000000,39.900000,13.070000,348.000000,1.000000,65.502500,4.000000,55.110000
50%,10.000000,24340.000000,1.000000,74.900000,16.250000,602.000000,1.000000,114.135000,5.000000,91.890000
75%,15.000000,59063.000000,1.000000,132.900000,21.140000,985.000000,3.000000,194.250000,5.000000,157.210000
max,100.000000,99980.000000,21.000000,1999.990000,196.940000,3992.000000,20.000000,7274.880000,5.000000,2115.050000


Podemos também utilizar a ferramenta Ydata-profiling para obter uma visualização geral dos dados - Ela faz análises simples mas muitos úteis em um primeiro momento com os dados. Para gerar a página HTML com os dados, você pode executar o programa em "src\etl\data_exploration.py" e abrir a página usando seu explorador de arquivos (ou como queira, é uma página web). A seguir detalho alguns pontos observados no conjunto de dados como um todo:

